tests/product/base_product_case.py

```diff
@@ -100,59 +97,51 @@ def detect_presto_rpm(self):
                                rpm_path)
             return rpm_filename
 
-    def setup_docker_cluster(self, cluster_type='centos'):
+    def setup_cluster(self, cluster_type='centos'):
         cluster_types = ['presto', 'centos']
-        if cluster_type not in cluster_types:
-            self.fail('{0} is not a supported cluster type. Must choose one'
-                      ' from {1}'.format(cluster_type, cluster_types))
-
         try:
-            are_presto_images_present = DockerCluster.check_for_presto_images()
-            if cluster_type == 'presto' and are_presto_images_present:
-                self.docker_cluster = DockerCluster.start_presto_cluster()
-                return
-            self.docker_cluster = DockerCluster.start_centos_cluster()
-            if cluster_type == 'presto' and not are_presto_images_present:
-                self.install_presto_admin(self.docker_cluster)
-                self.upload_topology()
-                self.server_install()
-                self.docker_client.commit(
-                    self.docker_cluster.master,
-                    INSTALLED_PRESTO_TEST_MASTER_IMAGE
-                )
-                self.docker_client.commit(
-                    self.docker_cluster.slaves[0],
-                    INSTALLED_PRESTO_TEST_SLAVE_IMAGE
-                )
+            if cluster_type == 'presto':
+                self.cluster = DockerCluster.start_presto_cluster(
+                    self.install_default_presto)
+            elif cluster_type == 'centos':
+                self.cluster = DockerCluster.start_centos_cluster()
+            else:
+                self.fail('{0} is not a supported cluster type. Must choose '
+                          'one from {1}'.format(cluster_type, cluster_types))
         except DockerClusterException as e:
             self.fail(e.msg)
 
+    def install_default_presto(self, cluster):
+        self.install_presto_admin(cluster=cluster)
+        self.upload_topology(cluster=cluster)
+        self.server_install(cluster=cluster)
+
     def tearDown(self):
         self.restore_stdout_stderr_keep_open()
-        if hasattr(locals()['self'], 'docker_cluster'):
-            self.docker_cluster.tear_down_containers()
+        if self.cluster:
```

Comment:
I added self.cluster = null at the top, so I think it'll work because it's always defined